### DISCUSSION TITLE
Is there a way to force the queue to process?

### DISCUSSION THREAD

**Discussion by zurtri** [django-q2/django-q2#80] [General]
When devving, sometimes the queue gets kind of "stuck" and doesn't process items.

I restart the queue manually, and this normally fixes it.

Is there a function I could call (or could be implemented) that tells django-q2 to process the currently queued items?

And thank you for forking from django-q - it is great to see this software back to being maintained.

Cheers
Andy

**Comment by GDay** [comment]
Hi Andy,

The issue here is probably that the process crashes (either due to a bug or due to the processes getting killed randomly), which might be related to this issue: https://github.com/django-q2/django-q2/issues/23

There kind of is a way to move them through the cluster manually, see this function:

https://github.com/django-q2/django-q2/blob/69a40344f002f622e7e15428a784db0cc204d315/django_q/tasks.py#L761-L777

I can't test it right now, but adding something like this is probably going to allow you to run through all pending tasks:

```
from django_q.models import OrmQ
from django_q.tasks import _sync

for task in OrmQ.objects.all():
    _sync(task.payload)
```

**Reply by zurtri** [reply]
Thank you mate. I'll give it a go.

**Comment by zurtri** [comment]
Yes this code seems to work as far as I can tell.

Also using `django-q2` - the queue doesnt get stuck as much as it did in `django-q`.

Thank you.
